src/main/js/PipelineEditorLink.jsx

```diff
@@ -46,6 +39,21 @@ class PipelineEditorLink extends React.Component {
             </Link>
         );
     }
+
+    @action
```

Comment:
this wasn't exactly what i meant by splitting this off to a separate `@action` method. Basically, whatever modifies the state of an `@observable` field needs to happen wrapped in `action(..)` or in a method annotated with `@action`